sbc_compassion/model/sponsorship_correspondance.py

```diff
@@ -0,0 +1,109 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014-2015 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Emmanuel Mathier <emmanuel.mathier@gmail.com>
+#
+#    The licence is in the file __openerp__.py
+#
+##############################################################################
+
+from openerp import fields, models, api, _
+
+
+class ResSponsorshipCorrespondence(models.Model):
+
+    """ This class create correspondances to match Compassion needs.
+    """
+
+    _name = 'sponsorship.correspondence'
+
+    ##########################################################################
+    #                                 FIELDS                                 #
+    ##########################################################################
+
+    sponsorship_id = fields.Many2one(
+        'recurring.contract', 'Sponsorship', required=True)
+    name = fields.Char('name', compute='_set_name')
+    partner_id = fields.Many2one(
+        related='sponsorship_id.correspondant_id', store=True)
+    child_id = fields.Many2one(related='sponsorship_id.child_id', store=True)
+    # Field used for identifying correspondence
+    kit_id = fields.Integer('Kit id', copy=False, readonly=True)
+    letter_type = fields.Selection(selection=[
+        ('S2B', _('Sponsor to beneficiary')),
+        ('B2S', _('Beneficiary to sponsor'))])
+    communication_type = fields.Selection(selection=[
+        ('scheduled', _('Scheduled')),
+        ('response', _('Response')),
+        ('thank_you', _('Thank you')),
+        ('third_party', _('Third party')),
+        ('christmas', _('Christmas')),
+        ('introduction', _('Introduction'))])
+    state = fields.Selection(selection=[
+        ('new', _('New')),
+        ('to_translate', _('To translate')),
+        ('translated', _('Translated')),
+        ('quality_checked', _('Quality checked')),
+        ('rejected', _('Rejected')),
+        ('sent', _('Sent')),
+        ('delivered', _('Delivered'))])
+    is_encourager = fields.Boolean()
+    mandatory_review = fields.Boolean(
+        related='sponsorship_id.correspondant_id.mandatory_review',
```

Comment:
I would not make a related field because if we set once for a letter `mandatory_review`, it will update in partner and then all its letters will be flagged for mandatory review. Instead, I would override create method to check on creation if the correspondent has this flag and set it in letter if needed.